File: server/internal/enum/enums.go
package enum

type MuscleGroup string

const (
	MuscleChest      MuscleGroup = "chest"
	MuscleBack       MuscleGroup = "back"
	MuscleShoulder   MuscleGroup = "shoulders"
	MuscleBiceps     MuscleGroup = "biceps"
	MuscleTriceps    MuscleGroup = "triceps"
	MuscleForearms   MuscleGroup = "forearms"
	MuscleAbs        MuscleGroup = "abs"
	MuscleGlutes     MuscleGroup = "glutes"
	MuscleQuads      MuscleGroup = "quads"
	MuscleHamstrings MuscleGroup = "hamstrings"
	MuscleCalves     MuscleGroup = "calves"
	MuscleFullBody   MuscleGroup = "full_body"
)

type Equipment string

const (
	EquipmentBarbell      Equipment = "barbell"
	EquipmentDumbbell     Equipment = "dumbbell"
	EquipmentMachine      Equipment = "machine"
	EquipmentCable        Equipment = "cable"
	EquipmentBodyweight   Equipment = "bodyweight"
	EquipmentBand         Equipment = "band"
	EquipmentKettlebell   Equipment = "kettlebell"
	EquipmentEZBar        Equipment = "ez_bar"
	EquipmentSmithMachine Equipment = "smith_machine"
	EquipmentOther        Equipment = "other"
)

type ExerciseCategory string

const (
	ExerciseCategoryStrength   ExerciseCategory = "strength"
	ExerciseCategoryCardio     ExerciseCategory = "cardio"
	ExerciseCategoryBodyweight ExerciseCategory = "bodyweight"
	ExerciseCategoryStretch    ExerciseCategory = "stretch"
)

type TrackingType string

const (
	TrackingTypeWeightReps TrackingType = "weight_reps"
	TrackingTypeTime       TrackingType = "time"
	TrackingTypeDistance   TrackingType = "distance"
	TrackingTypeRepsOnly   TrackingType = "reps_only"
)

type WeightUnit string

const (
	WeightUnitKG  WeightUnit = "kg"
	WeightUnitLBS WeightUnit = "lbs"
)

type PlanGoal string

const (
	PlanGoalHypertrophy PlanGoal = "hypertrophy"
	PlanGoalStrength    PlanGoal = "strength"
	PlanGoalEndurance   PlanGoal = "endurance"
	PlanGoalGeneral     PlanGoal = "general"
)

File: mobile/src/constants/enums.ts
export enum MuscleGroup {
  Chest = "chest",
  Back = "back",
  Shoulders = "shoulders",
  Biceps = "biceps",
  Triceps = "triceps",
  Forearms = "forearms",
  Abs = "abs",
  Glutes = "glutes",
  Quads = "quads",
  Hamstrings = "hamstrings",
  Calves = "calves",
  FullBody = "full_body",
}

export enum Equipment {
  Barbell = "barbell",
  Dumbbell = "dumbbell",
  Machine = "machine",
  Cable = "cable",
  Bodyweight = "bodyweight",
  Band = "band",
  Kettlebell = "kettlebell",
  EzBar = "ez_bar",
  SmithMachine = "smith_machine",
  Other = "other",
}

export enum ExerciseCategory {
  Strength = "strength",
  Cardio = "cardio",
  Bodyweight = "bodyweight",
  Stretch = "stretch",
}

export enum TrackingType {
  WeightReps = "weight_reps",
  Time = "time",
  Distance = "distance",
  RepsOnly = "reps_only",
}

export enum WeightUnit {
  KG = "kg",
  LBS = "lbs",
}

export enum PlanGoal {
  Hypertrophy = "hypertrophy",
  Strength = "strength",
  Endurance = "endurance",
  General = "general",
}

export enum PRType {
  MaxWeight = "max_weight",
  MaxVolume = "max_volume",
  Estimated1RM = "estimated_1rm",
}

export enum ChallengeType {
  Volume = "volume",
  Frequency = "frequency",
  TimeSlot = "time_slot",
  CardioDuration = "cardio_duration",
}

export enum CoachClientStatus {
  Active = "active",
  Paused = "paused",
  Terminated = "terminated",
}

export enum SharedPlanDifficulty {
  Beginner = "beginner",
  Intermediate = "intermediate",
  Advanced = "advanced",
}
